src/credentials/tests/TestDeviceAttestationConstruction.cpp

```diff
@@ -0,0 +1,159 @@
+
+        // test with missing mandatory TLV entries
+        err = ConstructAttestationElements(ByteSpan(), ByteSpan(attestationNonce), timestamp, ByteSpan(), vendorReservedArray,
+                                           ArraySize(vendorReservedArray), vendorId, profileNum, attestationElementsSpan);
+        NL_TEST_ASSERT(inSuite, err == CHIP_ERROR_INVALID_ARGUMENT);
+
+        // test with missing mandatory TLV entries
+        err =
+            ConstructAttestationElements(ByteSpan(certificationDeclaration), ByteSpan(), timestamp, ByteSpan(), vendorReservedArray,
+                                         ArraySize(vendorReservedArray), vendorId, profileNum, attestationElementsSpan);
+        NL_TEST_ASSERT(inSuite, err == CHIP_ERROR_INVALID_ARGUMENT);
+
+        err = ConstructAttestationElements(ByteSpan(certificationDeclaration), ByteSpan(attestationNonce), timestamp, ByteSpan(),
+                                           vendorReservedArray, ArraySize(vendorReservedArray), vendorId, profileNum,
+                                           attestationElementsSpan);
+        NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+        attestationElementsLen = attestationElementsSpan.size();
+    }
+
+    ByteSpan certificationDeclarationSpan;
+    ByteSpan attestationNonceSpan;
+    uint32_t timestampDeconstructed;
+    ByteSpan firmwareInfoSpan;
+    ByteSpan vendorReservedDeconstructed[2];
+    uint16_t vendorIdDeconstructed;
+    uint16_t profileNumDeconstructed;
+
+    err =
+        DeconstructAttestationElements(ByteSpan(attestationElements.Get(), attestationElementsLen), certificationDeclarationSpan,
+                                       attestationNonceSpan, timestampDeconstructed, firmwareInfoSpan, vendorReservedDeconstructed,
+                                       ArraySize(vendorReservedDeconstructed), vendorIdDeconstructed, profileNumDeconstructed);
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+
+    NL_TEST_ASSERT(inSuite, certificationDeclarationSpan.data_equal(ByteSpan(certificationDeclaration)));
+    NL_TEST_ASSERT(inSuite, attestationNonceSpan.data_equal(ByteSpan(attestationNonce)));
+    NL_TEST_ASSERT(inSuite, timestamp == timestampDeconstructed);
+    NL_TEST_ASSERT(inSuite, firmwareInfoSpan.empty());
+    NL_TEST_ASSERT(inSuite, ArraySize(vendorReservedArray) == ArraySize(vendorReservedDeconstructed));
+    for (size_t i = 0; i < ArraySize(vendorReservedArray); ++i)
+    {
+        NL_TEST_ASSERT(inSuite, vendorReservedArray[i].data_equal(vendorReservedDeconstructed[i]));
+    }
+    NL_TEST_ASSERT(inSuite, vendorIdDeconstructed == vendorId);
+    NL_TEST_ASSERT(inSuite, profileNumDeconstructed == profileNum);
+}
+
+/**
+ *   Test Suite. It lists all the test functions.
+ */
+// clang-format off
+static const nlTest sTests[] = {
+    NL_TEST_DEF("Test Device Attestation Elements Construction and Deconstruction", TestAttestationElements),
```

Comment:
Split test into construction/deconstruction. The test you currently have can be kept as a "RoundTrip" test, but we do need separate test cases as well for contruction+args --> expected octet string and octet string --> expected args.

Reply:
will update